core/pta/device.c

```diff
@@ -53,6 +56,25 @@ static TEE_Result get_devices(uint32_t types,
 		}
 	}
 
+	if (IS_ENABLED(CFG_EARLY_TA)) {
+		const struct early_ta *eta;
```

Comment:
why? const struct pseudo_ta_head *ta;  also is not init to NULL and there is for_each_early_ta(..) with take eta as variable.

Reply:
You're right that there is no issue with current implementation.
However, it is part of optee_os coding style that all local variables should get an initial value were defined:
https://optee.readthedocs.io/en/latest/general/coding_standards.html#coding-standards
This is best practice matters.

So here prefer `const struct early_ta *eta = NULL;`.
Compiler will wipe it out this since it sees `eta` effectively gets a initial value in `for_each_early_ta()` before is it used.